src/app/view/components/dot-large-message-display/dot-large-message-display.component.ts

```diff
@@ -2,21 +2,58 @@ import {
     DotLargeMessageDisplayService,
     DotLargeMessageDisplayParams
 } from './services/dot-large-message-display.service';
-import { Component, OnInit } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnInit, ViewChild, ElementRef, Renderer2, OnDestroy } from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
     selector: 'dot-large-message-display',
     templateUrl: './dot-large-message-display.component.html',
     styleUrls: ['./dot-large-message-display.component.scss']
 })
-export class DotLargeMessageDisplayComponent implements OnInit {
-    data$: Observable<DotLargeMessageDisplayParams>;
+export class DotLargeMessageDisplayComponent implements OnInit, OnDestroy {
+    @ViewChild('body') bodyEl: ElementRef;
 
-    constructor(public dotLargeMessageDisplayService: DotLargeMessageDisplayService) {}
+    data: DotLargeMessageDisplayParams;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private dotLargeMessageDisplayService: DotLargeMessageDisplayService,
+        private renderer: Renderer2
+    ) {}
 
     ngOnInit() {
-        this.data$ = this.dotLargeMessageDisplayService.sub();
+        this.dotLargeMessageDisplayService
+            .sub()
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((content: DotLargeMessageDisplayParams) => {
+                this.data = content;
+
+                if (content) {
+                    setTimeout(() => {
```

Comment:
why?

Reply:
Because the `this.body` doesn't exist, it have a `ngif` so we have to wait until render.

Reply:
https://angular.io/api/core/AfterViewInit ???

Reply:
In this case doesn't apply, because this is code that run in the subscribe, so the div shows and hide depending on the subscribe is not showing on load.